### ISSUE TITLE
Error- stage-4-start-pc2drc (CConn: unable connect to socket: Connection refused (111))

### ISSUE THREAD

**Issue by Special-Niewbie** [thefloppydriver/pc2drc#12]
**Describe the bug**
Once you have started `stage-3-pair-with-wiiu.py` and followed the whole procedure to connect the Wii U Pad to PC via the Wireless USB key.
Premise in my Desktop PC with the files `stage-3-pair-with-wiiu.py` and `stage-4-start-pc2drc`, gives me the same error as @Etunts as `Error: User directory is not / home / root` and i solved / I was able to move on with her modified files from @Etunts here: https://github.com/thefloppydriver/pc2drc/issues/9

So, once I pair the Wii U Pad to the PC and run the `stage-4-start-pc2drc.py` ,  I get the error `Could not connect to vncserver ERROR 111`.
PLease watch the txt file below:
[stage-4-start-pc2drc.txt](https://github.com/thefloppydriver/pc2drc/files/8386154/stage-4-start-pc2drc.txt)

**Some Tips**
The strange of the `Error: User directory is not / home / root`, with stage-3 and stage-4 I got only in my Desktop side because on my second Netbook computer with Ubuntu 20.04 LTS I didn't get this error and I can perform all the stages without mod it.
So, not sure if it's related to the OS version (because on Desktop sde I'm using the Ubuntu 18.04.6) or for some mysterious black magic...

**Desktop:**
 - OS: Ubuntu 18.04.6 LTS
 - Kernel Version: 5.4.0-107-generic
 - System Architecture: x86_64

**Comment by Etunts** [comment]
[usethesev2.zip](https://github.com/thefloppydriver/pc2drc/files/8398037/usethesev2.zip)

**Comment by Special-Niewbie** [comment]
Hi @Etunts , you mean that with this new fix you can connect to the PAD, because you fix the 111 Error? Am I understandand correctly?

**Comment by Etunts** [comment]
No. Actually this one shows more errors and it only fixes home/root error. I posted this one because its would be more easy to fix, the first one i sent silents the other errors, not fixing them.

**Comment by Etunts** [comment]
Silenting an error is not a solution, actually  it makes it more hard to debug things.

**Comment by Special-Niewbie** [comment]
I noticed that in these lines, was the user "thefloppydriver", and I changed with mine:
```

print("Starting VNC server...")
#subprocess.check_call(['tigervncserver', '--kill', ':1']) 

time.sleep(0.5)
#os.system("sudo runuser -u cris -- /bin/bash -c \"echo -e '12345678\\n12345678\\nn' | vncpasswd\"") #unnecessary
#subprocess.check_call(['chmod', '0664', user_dir+'/.vnc/passwd'])
#os.system("sudo runuser -u cris -- tigervncserver :1 -passwd "+user_dir+"/.vnc/passwd -depth 24 -geometry 640x480 -localhost yes")
try:
    os.system("sudo runuser -u cris -- tigervncserver :1 -useold -FrameRate 59.94 -SecurityTypes None -depth 24 -geometry 640x480 -localhost yes -xstartup "+current_working_directory+"/libdrc-vnc/vncconfig-files/Xvnc-session-gnome")
except:
    print("Failed to start tigervncserver, killing and trying again.")
    subprocess.check_call(['tigervncserver', '--kill', ':1'])
    try:
        os.system("sudo runuser -u cris -- tigervncserver :1 -useold -FrameRate 59.94 -SecurityTypes None -depth 24 -geometry 640x480 -localhost yes -xstartup "+current_working_directory+"/libdrc-vnc/vncconfig-files/Xvnc-session-gnome")
    except:
        print("Could not start tigervncserver.")
        input("(press enter to quit)")
        sys.exit(0)
```
and then the 111 error it doesn't come out, but it come out another error that it say:

```
During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "./stage-4-start-pc2drc_mod2.py", line 233, in <module>
    print(vncserver.before.decode('utf-8'))
NameError: name 'vncserver' is not defined

```
so, seems the vncserver it is not defined?

**Comment by Etunts** [comment]
I changed that vncserver to vncviewer which silented all of errors.

**Comment by Special-Niewbie** [comment]
> I changed that vncserver to vncviewer which silented all of errors.

I think there is the miss understanding here , if  I go back to the original file stage-4-start-pc2drc from @thefloppydriver , there in the line 236 `print(vncserver.before.decode('utf-8'))` and in your 233 `print(vncserver.before.decode('utf-8'))` they are completely the same from the beginning , so what you changed from the original file it's that you deleted completely the lines from 18 to 21 to fix the home/root error. So, now it's the problem that on the line 233 `print(vncserver.before.decode('utf-8'))` it miss or misspell the `vncserver` that it's not defined instead of the `vncviewer` it is, and that the point of the error.
But I don't know hot to solve it, @thefloppydriver please man , where are you? We need your help....

**Comment by Etunts** [comment]
He said this to me on reddit. Lets wait for a while.

Yes, but I'm focusing on school and moving houses right now. I'll return to it very soon. Thank you for your interest in the project :)

**Comment by Special-Niewbie** [comment]
> He said this to me on reddit. Lets wait for a while.
> 
> Yes, but I'm focusing on school and moving houses right now. I'll return to it very soon. Thank you for your interest in the project :)

Thank you for the info @Etunts , okay, I will be waiting any news here......

**Comment by gudenau** [comment]
The problem appears to be the fact that the scripts have a hardcoded username.

**Comment by gudenau** [comment]
The flags for `drcvncclient` are also missing `-lpthread` and `-ldl`, plus `drcvncclient` instantly exits.
